### ISSUE TITLE
Dragging histogram should not change viewer image

### ISSUE THREAD

**Issue by riversn** [bradmartin333/XferSuite#249]
When dragging the criteria histogram to zoom in/look at some areas, it shouldn't open a feature in the viewer when clicked there. only when clicking a bar and not initiating a drag or movement, should the viewer pop up and display the area of interest.

Also, if the viewer is minimized, it should get brought to front if a new histograph bar is clicked.

**Comment by bradmartin333** [comment]
> When dragging the criteria histogram to zoom in/look at some areas, it shouldn't open a feature in the viewer when clicked there. only when clicking a bar and not initiating a drag or movement, should the viewer pop up and display the area of interest.

Will look into it - lots of ways to handle this, but need to find the simplest one

> Also, if the viewer is minimized, it should get brought to front if a new histograph bar is clicked.

This should be the case already? On each click, any open viewer is closed a new one is made

**Comment by riversn** [comment]
It re-opens the 'new' viewer minimized, when it should be brought to the normal window size I think.

**Comment by bradmartin333** [comment]
> It re-opens the 'new' viewer minimized, when it should be brought to the normal window size I think.

I couldn't recreate this, but I did add something that should prevent it as well as keep the last location.
If that introduces more bugs, let me know and I will remove the location saving part of it.

**Comment by riversn** [comment]
Need to add the 'wiggle room' suggested. Some of the users have a hard time clicking and holding still.

Also, the cursor where the histogram clicks and where the cursor are, are actually slightly offset. It's like the user is pressing the middle of the bar in the bar graph, but the cursor actually clicks to the left of the bar. So the user has to offset the cursor to the right to compensate

**Comment by bradmartin333** [comment]
I just tried being pretty rough with it and didn't have those issues. Are you/the users on Win10 using a mouse?

**Comment by riversn** [comment]
No, I just tested on a small distribution, if you click directly on a bar, it just doesn't register that you're clicking it... you have to click on the left or right of the bar - in the white space, not the blue bar itself

**Comment by riversn** [comment]
If you click in the red area, it doesn't load the image viewer, but if you click to the sides, it does

**Comment by riversn** [comment]
Also, we do need to add the 'wiggle room' so that if the user clicks and moves a tiny bit, it still registers as a click.

> Done - it does a simple check if the cursor position at MouseDown is the same as the MouseUp trigger. If you have issues later on, a wiggle room buffer could be added so you can still move the mouse 1 pixel or two - let me know.

**Comment by bradmartin333** [comment]
can I add a "hold shift to prevent opening image scroller"?

I can try updating the plotting package, but I don't have the files to test each and every plot to make sure nothing else changes

**Comment by bradmartin333** [comment]
![demoFeatureOLVchangesMay2023](https://github.com/bradmartin333/XferSuite/assets/19335151/8fdc4008-a364-418a-95eb-66524a352d26)

idk... maybe I am missing something

**Comment by riversn** [comment]
Yeah, not exactly what I was trying to fix - 

> ![image](https://user-images.githubusercontent.com/101815602/238673713-b7272187-e086-4f55-8961-23dc174a8e52.png)

The issue is left clicking on a bar to view the image viewer for that sample. The clicks do not open the correct image. If you click on a bar directly nothing happens, but if you click to the left or right of the bar it works, but may not open the correct sample.

**Comment by riversn** [comment]
It only happens away from the main distribution really... and low count bars

**Comment by bradmartin333** [comment]
zoom/pan might help?
